src/classes/STG_PanelHouseholds_CTRL.cls

```diff
@@ -53,49 +53,66 @@ public with sharing class STG_PanelHouseholds_CTRL extends STG_Panel {
+    * @description Returns whether all relevant Settings fields are correct or not.
+    * Adds an error Page Message for any errors.
+    * @return boolean 
+    */
+    public Boolean isValidSettings() {
+        if (STG_Panel.stgService.stgHH.npo02__Household_Rules__c != HH_Households.NO_HOUSEHOLDS_PROCESSOR &&
+           STG_Panel.stgService.stgCon.npe01__Account_Processor__c == CAO_Constants.HH_ACCOUNT_PROCESSOR
+        ) {
+            addPageErrorMessage(system.Label.stgValidationHHAccountHHRules);
+            return false;
+        }
+
+        return isValidHouseholdNamingSettings();
+    }
     
     /*********************************************************************************************************
-    * @description Returns wwhether the custom name formats are valid or not, and adds a Page Message for
-    * any errors.
+    * @description Returns whether the Household Naming Settings fields are correct or not.
+    * Adds an error Page Message for any errors.
     * @return boolean 
     */
-    public boolean isValidNameFormats() {
-        // make sure any custom name formats are valid.
-        if (STG_Panel.stgService.stgHH.npo02__Advanced_Household_Naming__c) {
-            string strSetting;        
-            try {
-                strSetting = UTIL_Describe.getFieldLabel(UTIL_Namespace.StrTokenNSPrefix('Household_Naming_Settings__c'), UTIL_Namespace.StrTokenNSPrefix('Household_Name_Format__c'));
-                strNameSpecExample(STG_Panel.stgService.stgHN, 'Household_Name_Format__c', 2); 
-                strSetting = UTIL_Describe.getFieldLabel(UTIL_Namespace.StrTokenNSPrefix('Household_Naming_Settings__c'), UTIL_Namespace.StrTokenNSPrefix('Formal_Greeting_Format__c'));
-                strNameSpecExample(STG_Panel.stgService.stgHN, 'Formal_Greeting_Format__c', 2); 
-                strSetting = UTIL_Describe.getFieldLabel(UTIL_Namespace.StrTokenNSPrefix('Household_Naming_Settings__c'), UTIL_Namespace.StrTokenNSPrefix('Informal_Greeting_Format__c'));     
-                strNameSpecExample(STG_Panel.stgService.stgHN, 'Informal_Greeting_Format__c', 2);
-                return true; 
-            } catch (Exception ex) {
-                ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.ERROR, 
-                    string.format(label.stgErrorInvalidNameFormat, new string[]{strSetting, ex.getMessage()})));
-                return false;           
-            }        
-        } 
-        return true;    
+    private boolean isValidHouseholdNamingSettings() {
+        try {
+            HH_HouseholdNamingSettingValidator validator = new HH_HouseholdNamingSettingValidator();
+            HH_HouseholdNamingSettingValidator.Notification notification = validator.validate(STG_Panel.stgService.stgHN);
+
+            if (notification.isSuccess()) {
+                return true;
+            }
+
+            // Display the first error only?
```

Comment:
The validator was created to remove responsibility from the STG_PanelHouseholds_CTRL class to validate settings. It is up to the controller to call the validator to validate and if there a validation error, then display an error, all errors or raise an exception where required. The code was changed, but without impact to user experience since one error is still displayed. However, since validator collects all errors now, I will make the change to display them all.